src/vscode-dts/vscode.proposed.chatParticipantAdditions.d.ts

```diff
@@ -121,6 +121,14 @@ declare module 'vscode' {
 		}, options?: { status?: { description: string; kind: ChatResponseReferencePartStatusKind } });
 	}
 
+	export class ChatResponseMovePart {
```

Comment:
Is a response part the right place to do this, or would it make more sense on the participant response? You want to do it in the middle of the request? Also seems like it only makes sense for inline chat. If this is an internal thing that isn't on the finalization track, it should go in chatParticipantPrivate.d.ts.

Reply:
Oh, is this what was driving the test codeaction I saw today? Could you do this up front and trigger `vscode.editorChat.start` to start inline chat in the other editor?

Reply:
> Oh, is this what was driving the test codeaction I saw today? Could you do this up front and trigger vscode.editorChat.start to start inline chat in the other editor?

If test generation is triggered from the context menu or a code action, that's exactly what we do -- we have a command that opens a test file and starts an inline chat there.

However, a user may open inline chat in their file and invoke `/tests`, then we need a way to move inline chat from that editor into the test file so that it's easier to see test being streamed in. It needs to happen in the middle of the request. 

> If this is an internal thing that isn't on the finalization track, it should go in chatParticipantPrivate.d.ts.

Oh, thanks for the note. https://github.com/microsoft/vscode-copilot/issues/7489